### ISSUE TITLE
Provide a hardened systemd template

### ISSUE THREAD

**Issue by karlseguin** [timberio/vector#6555]
The provided systemd file (https://github.com/timberio/vector/blob/master/distribution/systemd/vector.service) gets a score of '9.6 unsafe" using `systemd-analyze security vector`

Consider providing a more secure default. I'm far from an expert in systemd and what capabilities vector needs (which is why I'd like to see a more secure default), but we've been running with:
```
UMask=077
NoNewPrivileges=yes
PrivateTmp=yes
ProtectSystem=full
ProtectHome=yes
ProtectControlGroups=yes
ProtectKernelModules=yes
ProtectKernelTunables=yes
RestrictNamespaces=yes
RestrictSUIDSGID=yes
LockPersonality=yes
PrivateDevices=yes
ProtectClock=yes
ProtectKernelLogs=yes
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6 AF_NETLINK
CapabilityBoundingSet=~CAP_SETUID CAP_SETGID CAP_SETPCAP CAP_SYS_ADMIN CAP_SYS_PTRACE CAP_NET_ADMIN CAP_RAWIO CAP_SYS_MODULE CAP_SYS_BOOT CAP_KILL
```

We also run with:
```
TemporaryFileSystem=/data:ro
BindPaths=/data/log
```

But this is definitely install-specific (mentioning it would be good, since it probably can't be part of the install template).

**Comment by jszwedko** [comment]
Hi @karlseguin I put together one over here: https://github.com/timberio/vector/pull/6696 . I'm curious what you make of it.

**Comment by karlseguin** [comment]
@jszwedko thanks, this really helped. Was able to use this as a baselines for getting it more secure.

I had to add `@system-service` to the `SystemCallFilter`. We glob files, and I think this is needed for `readdir`..but I think there's more than just `readdir`, so I added `@system-service`

I also, sadly, had to add `CAP_DAC_READ_SEARCH` to both `AmbientCapabilities` and `CapabilityBoundingSet`. Unfortunately, we have log files who's ownership is very hard to control (e.g. nginx). I feel like our only choices are (a) run vector as root  (b) run vector as `vector` and make it a member of a bunch of groups (nginx, postgres, ...)  (c) enable `CAP_DAC_READ_SEARCH`. I think (c) is the best option.

I don' think `CAP_DAC_READ_SEARCH` should be included by default, but I do think reading files owned by different users is probably a common issue people are reading on, and good documentation could help people avoid running vector as root.
